compiler/rustc_trait_selection/src/solve/mod.rs

```diff
@@ -275,6 +295,36 @@ impl<'a, 'tcx> EvalCtxt<'a, 'tcx> {
     ) -> QueryResult<'tcx> {
         self.make_canonical_response(Certainty::Yes)
     }
+
+    fn compute_subtype_goal(
+        &mut self,
+        goal: Goal<'tcx, SubtypePredicate<'tcx>>,
+    ) -> QueryResult<'tcx> {
+        self.infcx.probe(|_| {
+            let InferOk { value: (), obligations } = self
+                .infcx
+                .at(&ObligationCause::dummy(), goal.param_env)
+                .sub(goal.predicate.a, goal.predicate.b)?;
```

Comment:
I know you said subtyping in the solver is sketch but I literally don't know what else to do here 🤣

Reply:
:rofl: as long as its outside of **trait** solving :3

can add the check for `infer <: infer` here because rn we return overflow for `sub <: sub` even though we should return ambiguity.

Reply:
I added the `infer <: infer` check, but unlike [`InferCtxt::subtype_predicate`](https://github.com/rust-lang/rust/blob/a6269dad38e6ede0013ba3688099544833933c63/compiler/rustc_infer/src/infer/mod.rs#L990), I didn't register a sub in the type variable table, presumably because we don't actually do anything with it when canonicalizing a response.

But I did leave a fixme anyways.